compiler/rustc_metadata/src/creader.rs

```diff
@@ -157,6 +157,43 @@ impl<'a> std::fmt::Debug for CrateDump<'a> {
     }
 }
 
+/// Reason that a crate is being sourced as a dependency.
+#[derive(Clone, Copy)]
+enum CrateOrigin<'a> {
+    /// This crate was a dependency of another crate.
+    Dependency {
```

Comment:
Maybe `IndirectDependency`? All variants of this enum are for dependencies of the current crate.